core/dbt/context/providers.py

```diff
@@ -1427,6 +1427,20 @@ def this(self) -> Optional[RelationProxy]:
             return None
         return self.db_wrapper.Relation.create_from(self.config, self.model)
 
+    @contextproperty
+    def state_relation(self) -> Optional[RelationProxy]:
+        """
+        For commands which add information about this node's corresponding
+        production version (via a --state artifact), access the Relation
+        object for that stateful other
```

Comment:
other feels vague (well, really, just Latinate/French in phrasing :) ). "source node", etc.?